### PR TITLE
Check storage resize first

### PR DESCRIPTION
**Pull request by lujiajing1126**
Signed-off-by: Megrez Lu <lujiajing1126@gmail.com>

According to the last commit 306b1014bb0db97a7ed298bbe12cb22637978412, we will diff pvc objects.

But the following resize check would be unreachable.

Relate to issue #205

### CODE REVIEW

File: controllers/factory/expansion.go [hunk 1]
```diff
@@ -66,19 +66,20 @@ func wasCreatedSTS(ctx context.Context, rclient client.Client, pvcName string, n
 		if actualPVC == nil || newPVC == nil {
 			return false
 		}
+
+		if i := newPVC.Spec.Resources.Requests.Storage().Cmp(*actualPVC.Spec.Resources.Requests.Storage()); i != 0 {
+			log.Info("must re-recreate sts, its pvc claim was changed", "size-diff", i)
```
Comment (f41gh7) [inline controllers/factory/expansion.go:71]: I've realized, that  `Storage().Cmp()` - returns not a diff size, but indicator int - is it the same or not. 

 Will fix it later.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

File: controllers/factory/expansion.go (hunk 1 above)
Comment (lujiajing1126) [reply controllers/factory/expansion.go:71]: Thanks for pointing it out! Maybe I could send another fix later.

**Pull request merged**